ouroboros-consensus/src/Ouroboros/Consensus/Ledger/Query.hs

```diff
@@ -182,9 +214,22 @@ instance SameDepIndex (BlockQuery blk) => SameDepIndex (Query blk) where
     = Just Refl
   sameDepIndex GetSystemStart _
     = Nothing
+  sameDepIndex GetHeaderStateTip GetHeaderStateTip
+    = Just Refl
+  sameDepIndex GetHeaderStateTip _
+    = Nothing
 
 deriving instance Show (BlockQuery blk result) => Show (Query blk result)
 
+data HeaderStateTip blk = HeaderStateTip {
+      queriedTipSlotNo  :: !SlotNo
+    , queriedTipBlockNo :: !BlockNo
+    , queriedTipInfo    :: !(TipInfo blk)
```

Comment:
Would your use case let you simply use `HeaderHash` instead of `TipInfo` here? You can use the `tipInfoHash` method to get it.

That way, you wouldn't need the `Serialise (TipInfo blk)` constraints anywhere. (I haven't double-checked, but I'm assuming `RunNode blk` implies `Serialise (HeaderHash blk)` or else something as effective.)

If you do need `TipInfo`, then would adding the serialization constraint to the superclasses of `HasAnnTip` suffice?

```
class ( StandardHash blk
      , Show     (TipInfo blk)
      , Eq       (TipInfo blk)
      , NoThunks (TipInfo blk)
      , Serialise (TipInfo blk)
      ) => HasAnnTip blk where
```

Or perhaps adding it instead to `RunNode`?

(I'll need to better consider the context and revisit where similar constraints are pooled up in order to review such a change.)

Reply:
Your suggestion to use `tipInfoHash` works great!